apl-core/src/main/java/com/apollocurrency/aplwallet/apl/exchange/service/DexMatcherServiceImpl.java

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+
+package com.apollocurrency.aplwallet.apl.exchange.service;
+
+
+import com.apollocurrency.aplwallet.apl.core.app.EpochTime;
+import com.apollocurrency.aplwallet.apl.eth.utils.EthUtil;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexCurrencies;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexOffer;
+import com.apollocurrency.aplwallet.apl.exchange.model.DexOfferDBMatchingRequest;
+import com.apollocurrency.aplwallet.apl.exchange.model.OfferType;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Objects;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author Serhiy Lymar
+ */
+
+
+
+
+@Singleton
+public class DexMatcherServiceImpl implements IDexMatcherInterface {
+    
+    private static final Logger log = LoggerFactory.getLogger(DexMatcherServiceImpl.class);
+    DexService dexService;
+    private EpochTime epochTime;
+    private DexOffer lastMatchedOffer; 
```

Comment:
Is this for debugging?
State is dangerous for multi threading.

Reply:
sure, static volatile will be better since there is a read-only mode